packages/std/src/uuid.rs

```diff
@@ -0,0 +1,104 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use std::ops::Deref;
+use std::str::FromStr;
+use uuid as raw_uuid;
+
+use crate::{from_slice, to_vec};
+use crate::{Api, Env, StdResult, Storage};
+
+/// Uuid Provides a Uuid that can be used deterministically.
+/// Use internally Uuidv5 and NAMESPACE_OID.
+/// The name is combined with cahin id, contract address, block height, and increased sequential.
+#[derive(
+    Serialize, Deserialize, Copy, Clone, Default, Debug, PartialEq, Eq, PartialOrd, Ord, JsonSchema,
+)]
+pub struct Uuid(#[schemars(with = "String")] raw_uuid::Uuid);
+impl Uuid {
+    pub fn as_slice(&self) -> &[u8] {
+        &self.as_bytes()[0..16]
+    }
+
+    // Port the new_v5 implementation of uuid to use deps.api
+    // https://github.com/uuid-rs/uuid/blob/2d6c147bdfca9612263dd7e82e26155f7ef8bf32/src/v5.rs#L33
+    fn new_v5(api: &dyn Api, namespace: &Uuid, name: &[u8]) -> StdResult<Self> {
+        let inputs = [namespace.as_bytes(), name];
+        let buffer = api.sha1_calculate(&inputs)?;
+
+        let mut bytes = raw_uuid::Bytes::default();
+        bytes.copy_from_slice(&buffer[..16]);
+        let mut builder = raw_uuid::Builder::from_bytes(bytes);
+        builder
+            .set_variant(raw_uuid::Variant::RFC4122)
+            .set_version(raw_uuid::Version::Sha1);
+
+        Ok(Uuid(builder.into_uuid()))
+    }
+}
+
+const CONTRACT_UUID_SEQ_KEY: &[u8] = b"contract_uuid_seq";
+
+pub fn new_uuid(env: &Env, storage: &mut dyn Storage, api: &dyn Api) -> StdResult<Uuid> {
+    let raw_seq = storage.get(CONTRACT_UUID_SEQ_KEY);
+    let seq: u64 = match raw_seq {
+        Some(data) => from_slice(&data).unwrap(),
+        None => 0,
+    };
+    let next_seq: u64 = if seq < u64::MAX { seq + 1 } else { 0 };
```

Comment:
use `seq.wrapping_add(1)` instead of this.
https://doc.rust-lang.org/std/primitive.u64.html#method.wrapping_add